autosklearn/data/validation.py

```diff
@@ -303,25 +345,6 @@ def _check_and_encode_features(
         Interprets a Pandas
         Uses .iloc as a safe way to deal with pandas object
         """
-
-        # If there is a Nan, we cannot encode it due to a scikit learn limitation
-        if np.any(pd.isnull(X.dropna(axis='columns', how='all'))):
-            # Ignore all NaN columns, and if still a NaN
-            # Error out
-            raise ValueError("Categorical features in a dataframe cannot contain "
-                             "missing/NaN values. The OrdinalEncoder used by "
-                             "Auto-sklearn cannot handle this yet (due to a "
-                             "limitation on scikit-learn being addressed via: "
-                             "https://github.com/scikit-learn/scikit-learn/issues/17123)"
-                             )
-        elif np.any(pd.isnull(X)):
-            # After above check it means that if there is a NaN
-            # the whole column must be NaN
-            # Make sure it is numerical and let the pipeline handle it
-            for column in X.columns:
-                if X[column].isna().all():
-                    X[column] = pd.to_numeric(X[column])
-
         # Start with the features
         if hasattr(X, "iloc"):
```

Comment:
This is not part of the change, but if the input is only a dataframe, this if/else is not necessary, right?